### ISSUE TITLE
Different behavior between dev and build

### ISSUE THREAD

**Issue by MarcGodard** [canjs/can-stache-helpers#9]
I have the following helper:
```
stache.addHelper('isArray', (obj, options) => {
  if (obj && Array.isArray(obj.get())) {
    return options.fn(true)
  } else {
    return options.inverse(this)
  }
})
```

In my stache, I use it this way:
```
{{#isArray (scope.vm.editItem[itemName])}}
{{else}}
{{/isArray}}
```

In Dev (using donejs) the following works:
```
{{#isArray (scope.vm.editItem[itemName])}}
  {{../itemName}}
{{else}}
  {{../itemName}}
{{/isArray}}
```

In build production (using donejs) the following works:
```
{{#isArray (scope.vm.editItem[itemName])}}
  {{itemName}}
{{else}}
  {{../itemName}}
{{/isArray}}
```

This is so annoying. Am I doing something wrong?

**Comment by MarcGodard** [comment]
Changing it to this seems to be working well on dev and production. Looks like I mixed some ways of doing helpers.

```
stache.addHelper('isArray', (obj, options) => {
  if (obj && typeof obj.get().length !== 'undefined') {
    return options.fn()
  } else {
    return options.inverse()
  }
})
```

**Comment by justinbmeyer** [comment]
Weird. This is probably a bug in can-stache.

**Comment by MarcGodard** [comment]
Yes, my other helpers that had the `true` and `this` worked fine. But they might not have been using any sub variables. But I removed them anyways. It took me a while to figure this one out, why risk that on my other helpers if I forget about this... lol.

Perhaps a warning would suffice.

**Comment by cherifGsoul** [comment]
I created a [repo](https://github.com/cherifGsoul/stache-helper-demo) to demonstrate the issue.

I didn't see a difference between dev and prod, but when using `return options.fn()` and `return options.inverse()` (without parameters) a warning is thrown that `../itemName` could not be found.

@MarcGodard  could you take a look to see if the example is correct?

**Comment by MarcGodard** [comment]
@cherifGsoul Like I said, removing the `true` and `this` made it work. I still think this should be fixed in some way considering the time sink it was for me.

But https://github.com/cherifGsoul/stache-helper-demo/blob/master/src/items-list/items-list.js in your repo looks like what the cause was.
